perceval/runtime/job_group.py

```diff
@@ -0,0 +1,281 @@
+        # include the added job's info to dataset in the group
+        self._modify_job_dataset(job_info)
+
+    def _modify_job_dataset(self, updated_info: dict):
+        # modifies the recorded information when a new job is added
+        group_data = self._read_job_group_from_disk()
+
+        saved_job_info = group_data['job_group_data']
+        saved_job_info.append(updated_info)
+
+        # self._group_info['job_group_data'] = saved_job_info  # save changes in object
+        self._write_job_group_to_disk()  # save changes to disk
+
+    def progress(self):
+        """
+        Lists through all jobs in the group and returns a list of status
+        If a status is changed from existing in file, that entry is
+        updated with new information
+        """
+        status_jobs_in_group = []
+
+        for job_entry in self._group_info['job_group_data']:
+            if not job_entry['status'] in ['SUCCESS', None]:
+                rj = self._recreate_remote_job_from_stored_data(job_entry)
+                rj_status = rj.status()
+                if rj_status == 'SUCCESS':
+                    # remove metadata information if job ran successfully
+                    del job_entry['metadata']
+                job_entry['status'] = rj_status  # update current status
+
+            status_jobs_in_group.append(job_entry['status'])
+
+        self._write_job_group_to_disk()  # rewrites the group information on disk
+
+        # replace None with 'Not_sent' - more readable
+        status_list = [status if status is not None else "Not_sent" for status in status_jobs_in_group]
+        return status_list
+
+    def _recreate_remote_job_from_stored_data(self, job_entry) -> RemoteJob:
+        # returns a RemoteJob object recreated using its id and platform metadata
+        platform_metadata = job_entry['metadata']
+        user_token = platform_metadata['headers']['Authorization'].split(' ')[1]
+
+        rpc_handler = RPCHandler(platform_metadata['platform'],
+                                 platform_metadata['url'], user_token)
+
+        return RemoteJob.from_id(job_entry['id'], rpc_handler)
+
+    def run_parallel(self, **kwargs):
+        # todo: refine and implement
+        pass
+
+    def run_sequential(self, delay: float, **kwargs):
+        # todo : refine and implement
+        # if delay:
+            # run each after delay
+        pass
+
+    def rerun_failed(self):
+        pass
+        # todo: implement
+        # will be based on the API from Cloud side
+
+    @staticmethod
+    def delete_all_job_groups():
+        """
+        To delete all existing Job groups on disk
+        """
+        jgrp_dir_path = JobGroup._get_job_group_dir()
+        list_groups = JobGroup.list_saved_job_groups()
+        ps = PersistentData()
+        for each_file in list_groups:
+            ps.delete_file(os.path.join(jgrp_dir_path, each_file))
+
+    @staticmethod
+    def delete_job_group(filename: str):
```

Comment:
Again, you shouldn't expect users to know about files and especially file extensions. The signature here should just take a job group name, not a filename.